app/models/rubygem.rb

```diff
@@ -123,6 +123,19 @@ def owned_by?(user)
     ownerships.exists?(user_id: user.id)
   end
 
+  def most_recent_version
+    latest = versions.select(&:latest).sort_by(&:number)
```

Comment:
(with the caveat as I mentioned above, there may be rubygems with hundreds of versions where this breaks down)